lib/services/local/resource.go

```diff
@@ -0,0 +1,365 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package local
+
+import (
+	"context"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/trace"
+)
+
+// CreateResources attempts to dynamically create the supplied resources.
+// This function returns `trace.AlreadyExistsError` if one or more resources
+// would be overwritten, and `trace.NotImplementedError` if any resources
+// are of an unsupported type (see `ItemizeResource(...)`).
+//
+// NOTE: This function is non-atomic and performs no internal synchronization;
+// backend must be locked by caller when operating in parallel environment.
+func CreateResources(b backend.Backend, resources ...services.Resource) error {
```

Comment:
can you add context as a first parameter here? this way you can really pass context from somewhere else instead of using `context.TODO()` below